### ISSUE TITLE
Get `Semigroupal` from `Arrow` for free

### ISSUE THREAD

**Issue by kailuowang** [typelevel/cats#2069]
We could add the following to `Semigroupal`'s companion
```scala
  implicit def catsSemigroupalForArrow[F[_, _], A](implicit F: Arrow[F]): Semigroupal[F[A, ?]] = new Semigroupal[F[A, ?]] {
    def product[B, C](fb: F[A, B], fc: F[A, C]): F[A, (B, C)] =
      F.andThen(F.lift((x: A) => (x, x)), F.split(fb, fc))
  }
```
#2063 is adding a `merge` method, so this instance can just delegate to that. 
To test this we might need to create a new `case class MyFunc[A, B](f: A => B)` and an `Arrow` instance for it and then test the Semigroupal laws against it. 

I set the milestone to be 1.1, but it can certainly be done by 1.0

**Comment by LukaJCB** [comment]
Would this be a binary compatible change?

**Comment by kailuowang** [comment]
I think so.

**Comment by ceedubs** [comment]
A relevant question would be whether there are conceivable `F` types that form an `Arrow` but that we _wouldn't_ want to derive this implicit `Semigroupal` for. I can't think of anything that might fall into that realm, but it's probably worth thinking about.

**Comment by kailuowang** [comment]
@ceedubs are you suggesting making it explicit?

**Comment by ceedubs** [comment]
@kailuowang that was one thought that I had. But I don't know if there's really any reason to that. Just my usual paranoia :). I'll ponder this a bit and give others a chance to weigh in over the next day or two, and if nobody can think of a reason that this implicit def shouldn't exist, then maybe it's either good or mostly harmless.

**Comment by stephen-lazaro** [comment]
I can open a PR for this. I think you can reasonable strengthen it to `Apply` as well, after playing with it a bit. Given that that's a fair amount stronger, it might lend more weight to explicitness over implicitness here.

**Comment by LukaJCB** [comment]
Fixed by #2078
